compiler-plugin/src/main/kotlin/com/google/devtools/ksp/KotlinSymbolProcessingExtension.kt

```diff
@@ -204,6 +223,21 @@ abstract class AbstractKotlinSymbolProcessingExtension(val options: KspOptions,
     private fun KSPLogger.hasError(): Boolean {
         return (this as MessageCollectorBasedKSPLogger).recordedEvents.any { it.severity == CompilerMessageSeverity.ERROR || it.severity == CompilerMessageSeverity.EXCEPTION }
     }
+
+    private fun handleException(call: () -> Unit) {
+        try {
+            call()
+        } catch (e: Exception) {
```

Comment:
This alone will break things using [Kotlin Compile Testing](https://github.com/tschuchortdev/kotlin-compile-testing), where people throw `CompilationErrorException` to make `CLICompiler` return `COMPILATION_ERROR`.

Instead of returning `AnalysisResult.success`, we should return `AnalysisResult.compilationError` when there are errors.